### PR TITLE
Use JitPack for PreciousStones

### PR DESCRIPTION
**Pull request by Pugabyte**

### CODE REVIEW

**Comment by bensku** [comment]
Why?

**Comment by Pugabyte** [comment]
1. It's more update to date
2. I'm working on an addon and it's failing to find PreciousStones
3. Why not?

**Comment by Snow-Pyon** [comment]
You could use:
```
compile 'com.github.marcelo-mason:PreciousStones:1.12-SNAPSHOT'
```
to build the latest commit in that branch instead.

File: build.gradle [hunk 1]
```diff
@@ -53,7 +57,7 @@ dependencies {
 	compile 'com.google.code.findbugs:findbugs:2.0.3'
 	compile 'com.sk89q:worldguard:6.1.1-SNAPSHOT'
 	compile 'net.milkbowl.vault:Vault:1.6.6'
-	compile 'net.sacredlabyrinth.Phaed:PreciousStones:10.0.5'
+	compile 'com.github.marcelo-mason:PreciousStones:24e3d4b'
```
Comment (Snow-Pyon) [inline build.gradle]: Wouldn't it be better if you use `1.12-SNAPSHOT` as version? So it gets the latest commit from that branch.

Comment (Pugabyte) [reply build.gradle]: Done, thanks

**Comment by bensku** [comment]
Latest commit from a branch is ticking bomb. What if someone pushes a commit that does not compile? Or malicious code?

If plugin author does not upload something to their existing Maven repository, there is ptobably a good reason. Not everything you find in Git repos is reasonably stable. Not to mention that there may be experimental API changes which will never be actually present in release builds.

I will investigate if Maven repo of PreciousStones is down. That might be a good reason to actually use Jitpack.

**Comment by Pugabyte** [comment]
Then I will revert it back to the previous version? Using a maven repository is no more stable than using git, because a majority of people do not test their releases, and most publish automatically.

**Comment by bensku** [comment]
Snapshop releases from Git branches are not a good idea. Pointing to specific, somewhat stable commit * should* be ok, though. One does not update their dependencies automatically. Even if they're supposed to be stable releases, they may cause... Interesting issues.

@Snow-Pyon I do hope you don't use Jitpack to snapshot branches for Skript. Master branch is usually a bit broken if it is not right before or after a release. Sometimes more, sometimes less - you never know, so careful here...

**Comment by Snow-Pyon** [comment]
If I have to do an addon, I just use njol's repo as it's less hassle. I may use your repo if there is an actual API change that would break in newer version, but as of now, there are no API changes from 2.2 so, I'll stick to njol's repo.

**Comment by Pugabyte** [comment]
I use the latest version of precious stones, have for a long time, it is entirely stable.

**Comment by bensku** [comment]
By all means, do whatever you want for your own projects, @Pugabyte. But in this case, it is sadly not acceptable that I have to take the responsibility of watching PreciousStones git repository for alarming changes.

Your complaint about dependency not resolving was valid, though. I used Jitpack to point to commit for latest public release. Actually, exactly same thing that you originally did - thanks for it.

**Pull request closed without merging**